src/robotFactory.js

```diff
@@ -1,10 +1,99 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    if (step > 0) {
+      this.coords.y += step;
+    }
+
+    return this;
+  };
```

Comment:
In this task you do not need `if` check and return statements - in all move methods.
```suggestion
  goForward(step = 1) {
    this.coords.y += step;
  };
```